### PR TITLE
fix: Cleanup by sending a command and kill when dropped

### PR DESCRIPTION
**Pull request by Blu-J**

### CODE REVIEW

File: libs/js_engine/src/artifacts/loadModule.js [hunk 1]
```diff
@@ -54,8 +54,8 @@ const runDaemon = (
       waitPromise = waitPromise || Deno.core.opAsync("wait_command", await rpcId)
       return waitPromise
     },
-    async term() {
-      return Deno.core.opAsync("term_command", await rpcId)
+    async term(signal = 15) {
```
Comment (dr-bonez) [inline libs/js_engine/src/artifacts/loadModule.js:57]:
```suggestion
    async signal(signal = 15) {
```

File: libs/js_engine/src/artifacts/loadModule.js [hunk 2]
```diff
@@ -54,8 +54,8 @@ const runDaemon = (
...
+      return Deno.core.opAsync("term_command", await rpcId, 15)
```
Comment (dr-bonez) [inline libs/js_engine/src/artifacts/loadModule.js:58]:
```suggestion
      return Deno.core.opAsync("send_signal", await rpcId, 15)
```

File: libs/js_engine/src/lib.rs [hunk 3]
```diff
@@ -97,7 +97,7 @@ struct JsContext {
     input: Value,
     variable_args: Vec<serde_json::Value>,
     command_inserter: ExecCommand,
-    term_command: TermCommand,
+    term_command: SendKillSignal,
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:100]:
```suggestion
    send_signal: SendKillSignal,
```

File: libs/js_engine/src/lib.rs [hunk 4]
```diff
@@ -187,7 +187,7 @@ pub struct JsExecutionEnvironment {
     version: Version,
     volumes: Arc<dyn PathForVolumeId>,
     command_inserter: ExecCommand,
-    term_command: TermCommand,
+    term_command: SendKillSignal,
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:190]:
```suggestion
    send_signal: SendKillSignal,
```

File: libs/js_engine/src/lib.rs [hunk 5]
```diff
@@ -197,7 +197,7 @@ impl JsExecutionEnvironment {
         version: &Version,
         volumes: Box<dyn PathForVolumeId>,
         command_inserter: ExecCommand,
-        term_command: TermCommand,
+        term_command: SendKillSignal,
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:200]:
```suggestion
        send_signal: SendKillSignal,
```

File: libs/js_engine/src/lib.rs [hunk 6]
```diff
@@ -930,13 +930,17 @@ mod fns {
     }
 
     #[op]
-    async fn term_command(state: Rc<RefCell<OpState>>, id: u32) -> Result<(), AnyError> {
-        let term_command_impl: TermCommand = {
+    async fn term_command(
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:933]:
```suggestion
    async fn send_signal(
```

File: libs/js_engine/src/lib.rs [hunk 7]
```diff
@@ -930,13 +930,17 @@ mod fns {
...
+        state: Rc<RefCell<OpState>>,
+        id: u32,
+        signal: u32,
+    ) -> Result<(), AnyError> {
+        let term_command_impl: SendKillSignal = {
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:938]:
```suggestion
        let send_signal_impl: SendKillSignal = {
```

File: libs/js_engine/src/lib.rs [hunk 8]
```diff
@@ -930,13 +930,17 @@ mod fns {
...
             let state = state.borrow();
             let ctx = state.borrow::<JsContext>();
             ctx.term_command.clone()
         };
-        if let Err(err) = term_command_impl(embassy_container_init::RpcId::UInt(id)).await {
+        if let Err(err) = term_command_impl(embassy_container_init::RpcId::UInt(id), signal).await {
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:943]:
```suggestion
        if let Err(err) = send_signal_impl(embassy_container_init::RpcId::UInt(id), signal).await {
```

File: backend/src/manager/mod.rs [hunk 9]
```diff
@@ -405,8 +405,8 @@ impl Manager {
     pub fn exec_command(&self) -> ExecCommand {
         self.persistant_container.exec_command()
     }
-    pub fn term_command(&self) -> TermCommand {
-        self.persistant_container.term_command()
+    pub fn term_command(&self) -> SendKillSignal {
```
Comment (dr-bonez) [inline backend/src/manager/mod.rs:408]:
```suggestion
    pub fn send_signal(&self) -> SendKillSignal {
```

File: libs/js_engine/src/lib.rs [hunk 10]
```diff
@@ -930,13 +930,17 @@ mod fns {
...
-        let term_command_impl: TermCommand = {
+    async fn term_command(
+        state: Rc<RefCell<OpState>>,
+        id: u32,
+        signal: u32,
+    ) -> Result<(), AnyError> {
+        let term_command_impl: SendKillSignal = {
             let state = state.borrow();
             let ctx = state.borrow::<JsContext>();
             ctx.term_command.clone()
```
Comment (dr-bonez) [inline libs/js_engine/src/lib.rs:941]:
```suggestion
            ctx.send_signal.clone()
```

File: backend/src/procedure/js_scripts.rs [hunk 11]
```diff
@@ -65,7 +65,7 @@ impl JsProcedure {
         input: Option<I>,
         timeout: Option<Duration>,
         exec_command: ExecCommand,
-        term_command: TermCommand,
+        term_command: SendKillSignal,
```
Comment (dr-bonez) [inline backend/src/procedure/js_scripts.rs:68]:
```suggestion
        send_signal: SendKillSignal,
```

**Review by dr-bonez — COMMENTED** [review]
just some renaming suggestions

File: backend/src/manager/mod.rs [hunk 12]
```diff
@@ -153,26 +149,30 @@ impl ManagerMap {
...
-    on_stop: Sender<OnStop>,
     manifest: Manifest,
     container_name: String,
     tor_keys: BTreeMap<InterfaceId, TorSecretKeyV3>,
+}
+
+pub struct ManagerSharedState {
+    seed: Arc<ManagerSeed>,
+    persistent_container: Option<PersistentContainer>,
+    status: (Sender<Status>, Receiver<Status>),
```
Comment (Blu-J) [inline backend/src/manager/mod.rs:170]: I believe that we could just keep the sender

File: .github/workflows/debian.yaml [hunk 13]
```diff
@@ -20,7 +20,7 @@ jobs:
       with:
         path: embassyos-0.3.x
     - run: |
-        cp -r debian embassyos-$(VERSION)/
```
Comment (Blu-J) [inline .github/workflows/debian.yaml]: Did you mean this?

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**